dali/test/python/test_internals_operator_external_source.py

```diff
@@ -0,0 +1,489 @@
+from nvidia.dali.pipeline import Pipeline
+import nvidia.dali.ops as ops
+import nvidia.dali.fn as fn
+import nvidia.dali.types as types
+import nvidia.dali.backend as backend
+import numpy as np
+from nose.tools import assert_raises
+from test_utils import check_output
+import random
+from collections import Iterable
+datapy = np
+
+cupy_used = False
+
+def use_cupy():
+    global cp
+    global cupy_used
+    global datapy
+    import cupy as cp
+    cupy_used = True
+    datapy = cp
```

Comment:
I think this file requires explanation at the top (maybe same for the ones that actually execute this code), that we use data as either nupy or cupy to test CPU and GPU inputs.

Reply:
Done